core/src/test/java/com/linecorp/armeria/server/Http1ServerDelayedCloseConnectionTest.java

```diff
@@ -0,0 +1,150 @@
+import java.io.PrintWriter;
+import java.net.BindException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.time.Duration;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+class Http1ServerDelayedCloseConnectionTest {
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.idleTimeoutMillis(0);
+            sb.http(0);
+            sb.https(0);
+            sb.tlsSelfSigned();
+            sb.service("/close", (ctx, req) -> {
+                return HttpResponse.builder()
+                                   .ok()
+                                   .content("OK\n")
+                                   .header(HttpHeaderNames.CONNECTION, "close")
+                                   .build();
+            });
+        }
+    };
+
+    @Test
+    void shouldDelayDisconnectByServerSideIfClientDoesNotHandleConnectionClose() throws IOException {
+        try (Socket socket = new Socket("127.0.0.1", server.httpPort())) {
+            socket.setSoTimeout(100000);
+            final int socketPort = socket.getLocalPort();
+            final PrintWriter writer = new PrintWriter(socket.getOutputStream());
+            writer.print("GET /close" + " HTTP/1.1\r\n");
+            writer.print("\r\n");
+            writer.flush();
+
+            final BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            assertThat(in.readLine()).isEqualTo("HTTP/1.1 200 OK");
+
+            String line;
+            boolean hasConnectionClose = false;
+            while ((line = in.readLine()) != null) {
+                if ("connection: close".equalsIgnoreCase(line)) {
+                    hasConnectionClose = true;
+                }
+                if (line.isEmpty() || line.contains(":")) {
+                    continue;
+                }
+                if (line.startsWith("OK")) {
+                    break;
+                }
+            }
+            final long readStartTimestamp = System.nanoTime();
+            final int readResult = in.read();
+            final long readDurationMillis = Duration.ofNanos(System.nanoTime() - readStartTimestamp).toMillis();
+
+            assertThat(hasConnectionClose).isTrue();
+            assertThat(readResult).isEqualTo(-1);
+
+            final long defaultHttp1ConnectionCloseDelayMillis = Flags.defaultHttp1ConnectionCloseDelayMillis();
+            assertThat(readDurationMillis).isBetween(
+                    defaultHttp1ConnectionCloseDelayMillis - 100,
+                    defaultHttp1ConnectionCloseDelayMillis + 1000
```

Comment:
nit: Let's use higher values to avoid introducing another flaky test. 
```suggestion
                    defaultHttp1ConnectionCloseDelayMillis - 2000,
                    defaultHttp1ConnectionCloseDelayMillis + 2000
```

Reply:
It will be safer 👍  I chaged it!